src/wasm-lib/kcl/src/ast/types/execute.rs

```diff
@@ -243,6 +243,34 @@ impl BinaryExpression {
             }
         }
 
+        // Check for logical operators.
+        if self.operator == BinaryOperator::And || self.operator == BinaryOperator::Or {
+            let left = json_as_bool(&left_json_value);
+            let right = json_as_bool(&right_json_value);
+
+            // If either left or right is None, we must return a KclError.
+            if left.is_none() || right.is_none() {
```

Comment:
Rather than check-and-unwrap, I'd suggest using a `let else` block to do them both at once, it'll be less likely to cause problems later.

See https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=3e7b0e1289bd08b404e643dda90aec15 for an example.

Reply:
made this change.  I get how this is better, will use this pattern in the future, thanks!